### PR TITLE
[PROCS-3709] Add support to run process checks in core agent

### PR DESCRIPTION
**Pull request by daniel-taf**
#### What this PR does / why we need it:
This PR adds initial support for running Processes-related checks in the core agent instead of the process-agent. This is documented as an experimental feature. 

#### Which issue this PR fixes
N/A

#### Special notes for your reviewer:
There is a possible misconfiguration: If `processAgent.enabled=true` and the process checks are moved to core agent without other running checks on the core agent, the process agent container enters crash loop. We advise in the documentation to set `processAgent.enabled=false` in that case.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Chart Version bumped
- [x] Documentation has been updated with helm-docs (run: `.github/helm-docs.sh`)
- [x] `CHANGELOG.md` has been updated
- [x] Variables are documented in the `README.md`

### CODE REVIEW

File: charts/datadog/README.md [hunk 1]
```diff
@@ -775,6 +775,7 @@ helm install <RELEASE_NAME> \
 | datadog.processAgent.enabled | bool | `true` | Set this to true to enable live process monitoring agent |
 | datadog.processAgent.processCollection | bool | `false` | Set this to true to enable process collection in process monitoring agent |
 | datadog.processAgent.processDiscovery | bool | `true` | Enables or disables autodiscovery of integrations |
+| datadog.processAgent.runInCoreAgent | bool | `false` | Set this to true to run the following features in the core agent: Live Processes, Live Containers, Process Discovery. # This is an experimental feature requiring Agent 7.53.0+ and Linux. Currently not compatible with APM Single Step Instrumentation. # If Orchestrator Explorer, Network Performance Monitoring is not being used, processAgent.enabled should be set to false to remove the process-agent container. |
```
Comment (robertjli) [inline charts/datadog/README.md:778]: orchestrator is now a core check, it doesn't run in the process agent

File: charts/datadog/templates/_processes-common-env.yaml [hunk 2]
```diff
@@ -0,0 +1,18 @@
+
```
Comment (robertjli) [inline charts/datadog/templates/_processes-common-env.yaml:1]: Is this blank line intentional?

**Review by robertjli — APPROVED** [review]
A few updates, but good to go afterward

File: charts/datadog/values.yaml [hunk 3]
```diff
@@ -624,6 +624,11 @@ datadog:
     # datadog.processAgent.processDiscovery -- Enables or disables autodiscovery of integrations
     processDiscovery: true
 
+    # datadog.processAgent.runInCoreAgent -- Set this to true to run the following features in the core agent: Live Processes, Live Containers, Process Discovery.
+    ## This is an experimental feature requiring Agent 7.53.0+ and Linux. Currently not compatible with APM Single Step Instrumentation.
+    ## If Orchestrator Explorer, Network Performance Monitoring is not being used, processAgent.enabled should be set to false to remove the process-agent container.
```
Comment (celenechang) [inline charts/datadog/values.yaml:629]: to clarify, if a user is using NPM and `runInCoreAgent: true`, then `processAgent.enabled` still should be `true` ?

File: charts/datadog/templates/_container-agent.yaml [hunk 4]
```diff
@@ -58,6 +58,7 @@
   env:
     {{- include "containers-common-env" . | nindent 4 }}
     {{- include "fips-envvar" . | nindent 4 }}
+    {{- include "processes-common-envs" . | nindent 4 }}
```
Comment (wiyu) [inline charts/datadog/templates/_container-agent.yaml:61]: Any reason not to guard this with a check of `runInCoreAgent`?

File: charts/datadog/README.md (hunk 1 above)
Comment (daniel-taf) [reply charts/datadog/README.md:778]: I was thinking about old versions where the orchestrator was still on the process agent, but that is < 7.51.0 so it shouldn't actually be an issue. I removed it from the description here.

File: charts/datadog/values.yaml (hunk 3 above)
Comment (daniel-taf) [reply charts/datadog/values.yaml]: Yes

File: charts/datadog/templates/_container-agent.yaml (hunk 4 above)
Comment (daniel-taf) [reply charts/datadog/templates/_container-agent.yaml:62]: Added it

Comment (celenechang) [reply charts/datadog/templates/_container-agent.yaml:62]: sorry i think we actually want to keep this as it was, for the effort to have all container configs replicated in the core agent

Comment (daniel-taf) [reply charts/datadog/templates/_container-agent.yaml:61]: No worries, I removed the guard.

**Review by celenechang — APPROVED** [review]

**Pull request merged**